<launch>
  <include file="$(find hrl_pr2_lib)/launch/openni_kinect_polled.launch"/>
  <node name="visual_servo_node" pkg="visual_servo" type="visual_servo_node" respawn="false" output="screen">
    <remap from="color_image_topic" to="/kinect_head/rgb/image_color"/>
    <remap from="depth_image_topic" to="/kinect_head/depth/image"/>
    <remap from="point_cloud_topic" to="/kinect_head/depth/points"/>

    <param name="tape_hue_value" value="15" type="int"/>
    <param name="tape_hue_threshold" value="50" type="int"/>
    <param name="hand_hue_value" value="120" type="int"/>
    <param name="hand_hue_threshold" value="10" type="int"/>
    <param name="default_sat_bot_value" value="100" type="int"/>
    <param name="default_sat_top_value" value="220" type="int"/>
    <param name="default_val_value" value="255" type="int"/>
 
    <param name="display_wait_ms" value="3" type="int"/>
    <param name="autostart_pcl_segmentation" value="false" type="bool"/>
    <param name="use_guided_pushes" value="true" type="bool"/>

    <param name="crop_min_x" value="22" type="int"/>
    <param name="crop_max_x" value="593" type="int"/>
    <param name="crop_min_y" value="32" type="int"/>
    <param name="crop_max_y" value="469" type="int"/>

    <param name="workspace_frame" value="/torso_lift_link" type="string"/>

    <param name="min_table_z" value="-1.0" type="double"/>
    <param name="max_table_z" value="1.0" type="double"/>
    <param name="min_pushing_y" value="-1.0" type="double"/>
    <param name="max_pushing_y" value="1.0" type="double"/>
    <param name="min_pushing_x" value="0.0" type="double"/>
    <param name="max_pushing_x" value="2.0" type="double"/>

    <param name="use_weighted_edges" value="false" type="bool"/>
    <param name="threshold_edges" value="true" type="bool"/>
    <param name="edge_weight_thresh" value="0.1" type="double"/>
    <param name="depth_edge_weight_thresh" value="0.05" type="double"/>
    <param name="depth_edge_weight" value="0.75" type="double"/>

    <param name="boundary_ransac_thresh" value="0.05" type="double"/>
    <param name="min_edge_length" value="4" type="int"/>
  </node>
</launch>
